api/catalog/api/views/image_views.py

```diff
@@ -132,7 +132,18 @@ def watermark(self, request, *_, **__):
         params.is_valid(raise_exception=True)
 
         image = self.get_object()
-        image_url = image.url
+
+        # initialising the image url to None by default
+        image_url = None
```

Comment:
Since `image_url` is assigned in both the `if` and `else` branches below, there is no need to initialise it with `None`.